From 448a724135c0fff5e559492b7b4e9a244361f205 Mon Sep 17 00:00:00 2001
From: Max Horn <max@quendi.de>
Date: Thu, 14 Apr 2022 11:45:17 +0200
Subject: [PATCH] kernel: don't waitpid(-1)

This causes problems when using GAP as a library. It also
shouldn't be necessary as we record a list of all known child
processes.
---
 src/iostream.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/iostream.c b/src/iostream.c
index 6289ab6ec..37723ebac 100644
--- a/src/iostream.c
+++ b/src/iostream.c
@@ -280,13 +280,6 @@ static void ChildStatusChanged(int whichsig)
     HashUnlock(PtyIOStreams);
 
 #if !defined(HPCGAP)
-    /* Collect up any other zombie children */
-    do {
-        retcode = waitpid(-1, &status, WNOHANG);
-        if (retcode == -1 && errno != ECHILD)
-            Pr("#E Unexpected waitpid error %d\n", errno, 0);
-    } while (retcode != 0 && retcode != -1);
-
     signal(SIGCHLD, ChildStatusChanged);
 #endif
 }
-- 
2.35.1

